openlibrary/plugins/upstream/utils.py

```diff
@@ -24,6 +24,7 @@
     urlencode as parse_urlencode,
     urlparse,
     urlunparse,
+    quote
```

Comment:
Oh, I see you quoted it up here; it's generally better to apply escaping as close to where it's needed. `Request` is a generic class with information about the current request ; it doesn't know what the down stream client is going to use `fullpath` for, so it shouldn't assume it needs to be quoted!
```suggestion
```